### PR TITLE
frontend: leverage theme palette for workflows components

### PR DESCRIPTION
**Pull request by lucechal14**
Replace hardcoded colors with theme palette references for components in worklows

Testing Performed
manual, unit tests

FYI for reviewers, this PR mainly aims to update components within frontend/workflows, while updating all instances of widespread colors as to avoid possible mismatches from working them on a directory basis. Odd instances of hardcoded colors elsewhere will be addressed in future PRs.

### CODE REVIEW

File: frontend/packages/core/src/icon.tsx [hunk 1]
```diff
@@ -43,12 +44,12 @@ export const StatusIcon: React.FC<StatusProps> = ({
       )}
       {variant === "success" && (
         <>
-          <StyledStatusIcon data-color="#69F0AE" /> {children}
+          <StyledStatusIcon data-color={theme.palette.success[200]} /> {children}
```
Comment (lucechal14) [inline frontend/packages/core/src/icon.tsx:47]: Screenshot 2023-12-19 at 10 44 58 a m

File: frontend/workflows/audit/src/logs/event-row.tsx [hunk 2]
```diff
@@ -16,17 +16,17 @@ import FileSaver from "file-saver";
...
 const ENDPOINT = "/v1/audit/getEvents";
 const COLUMN_COUNT = 6;
-const MonospaceText = styled("div")({
+const MonospaceText = styled("div")(({ theme }) => ({
   fontFamily: "monospace",
   padding: "8px",
   border: "1px solid lightgray",
   borderRadius: "8px",
-  background: "#ddd9d9",
+  background: theme.palette.secondary[200],
```
Comment (lucechal14) [inline frontend/workflows/audit/src/logs/event-row.tsx:24]: Screenshot 2023-12-19 at 10 47 09 a m

File: frontend/workflows/envoy/src/remote-triage/clusters.tsx [hunk 3]
```diff
@@ -27,11 +28,16 @@ interface RatioStatusProps {
...
 
 const RatioStatus: React.FC<RatioStatusProps> = ({ succeeded, failed }) => {
+  const theme = useTheme();
   const total = succeeded + failed;
   return (
     <>
-      {succeeded !== 0 && <Bar fill="#69F0AE" width={`${(succeeded / total) * 100}px`} />}
-      {failed !== 0 && <Bar fill="#FF8A80" width={`${(failed / total) * 100}px`} />}
+      {succeeded !== 0 && (
+        <Bar fill={theme.palette.success[200]} width={`${(succeeded / total) * 100}px`} />
```
Comment (lucechal14) [inline frontend/workflows/envoy/src/remote-triage/clusters.tsx:36]: Screenshot 2023-12-19 at 10 48 24 a m

File: frontend/workflows/envoy/src/remote-triage/clusters.tsx [hunk 4]
```diff
@@ -27,11 +28,16 @@ interface RatioStatusProps {
...
+      )}
+      {failed !== 0 && (
+        <Bar fill={theme.palette.error[600]} width={`${(failed / total) * 100}px`} />
```
Comment (lucechal14) [inline frontend/workflows/envoy/src/remote-triage/clusters.tsx:39]: Screenshot 2023-12-19 at 10 49 51 a m

File: frontend/workflows/envoy/src/remote-triage/index.tsx [hunk 5]
```diff
@@ -82,12 +84,12 @@ const TriageDetails: React.FC<WizardChild> = () => {
     {
       id: "Running",
       value: healthyClusterCount,
-      color: "#69F0AE",
+      color: theme.palette.success[200],
```
Comment (lucechal14) [inline frontend/workflows/envoy/src/remote-triage/index.tsx:87]: Screenshot 2023-12-19 at 10 50 23 a m

File: frontend/workflows/envoy/src/remote-triage/index.tsx [hunk 6]
```diff
@@ -82,12 +84,12 @@ const TriageDetails: React.FC<WizardChild> = () => {
...
     },
     {
       id: "Failing",
       value: failingClusterCount,
-      color: "#FF8A80",
+      color: theme.palette.error[600],
```
Comment (lucechal14) [inline frontend/workflows/envoy/src/remote-triage/index.tsx:92]: Screenshot 2023-12-19 at 10 50 59 a m

File: frontend/workflows/projectCatalog/src/catalog/index.tsx [hunk 7]
```diff
@@ -119,14 +120,14 @@ const Catalog: React.FC<WorkflowProps> = () => {
   return (
     <Box style={{ padding: "32px" }}>
       <div style={{ marginBottom: "8px" }}>
-        <Typography variant="caption2" color="rgb(13, 16, 48, .48)">
+        <Typography variant="caption2" color={alpha(theme.palette.secondary[900], 0.48)}>
```
Comment (lucechal14) [inline frontend/workflows/projectCatalog/src/catalog/index.tsx:123]: this is a perfect match, rgb(13,16,48) === #0d1030 === secondary[900]

File: frontend/workflows/projectCatalog/src/catalog/index.tsx [hunk 8]
```diff
@@ -119,14 +120,14 @@ const Catalog: React.FC<WorkflowProps> = () => {
...
           Project Catalog&nbsp;/&nbsp;Index
         </Typography>
       </div>
       <div style={{ marginBottom: "32px" }}>
         <Typography variant="h2">Project Catalog</Typography>
         <div style={{ marginTop: "8px" }}>
-          <Typography variant="subtitle3" color="rgb(13, 16, 48, .48)">
+          <Typography variant="subtitle3" color={alpha(theme.palette.secondary[900], 0.48)}>
```
Comment (lucechal14) [inline frontend/workflows/projectCatalog/src/catalog/index.tsx:130]: this is a perfect match, rgb(13,16,48) === #0d1030 === secondary[900]

File: frontend/workflows/projectCatalog/src/details/card.tsx [hunk 9]
```diff
@@ -55,7 +55,7 @@ const StyledProgressContainer = styled("div")(({ theme }) => ({
   marginTop: "-12px",
   height: "4px",
   ".MuiLinearProgress-root": {
-    backgroundColor: "rgb(194, 200, 242)",
+    backgroundColor: theme.palette.primary[400],
```
Comment (lucechal14) [inline frontend/workflows/projectCatalog/src/details/card.tsx:58]: this is a perfect match, rgb(194,200,242) === #c2c8f2 === primary[400]

File: frontend/workflows/projectCatalog/src/details/header.tsx [hunk 10]
```diff
@@ -41,7 +42,11 @@ const BreadCrumbs = ({ routes = [] }: BreadCrumbProps) => {
     }
 
     return (
-      <StyledCrumb variant="caption2" color="rgb(13, 16, 48, .48)" key={route.title}>
+      <StyledCrumb
+        variant="caption2"
+        color={alpha(theme.palette.secondary[900], 0.48)}
```
Comment (lucechal14) [inline frontend/workflows/projectCatalog/src/details/header.tsx:47]: this is a perfect match, rgb(13,16,48) === #0d1030 === secondary[900]

File: frontend/workflows/projectSelector/src/card.tsx [hunk 11]
```diff
@@ -16,7 +16,7 @@ import { LinearProgress } from "@mui/material";
 const StyledProgressContainer = styled("div")(({ theme }) => ({
   height: "4px",
   ".MuiLinearProgress-root": {
-    backgroundColor: "rgb(194, 200, 242)",
+    backgroundColor: theme.palette.primary[400],
```
Comment (lucechal14) [inline frontend/workflows/projectSelector/src/card.tsx:19]: this is a perfect match, rgb(194,200,242) === #c2c8f2 === primary[400]

File: frontend/workflows/projectSelector/src/project-group.tsx [hunk 12]
```diff
@@ -94,10 +94,10 @@ const StyledClearIcon = styled("span")(({ theme }) => ({
     color: alpha(theme.palette.secondary[900], 0.38),
   },
   ".MuiIconButton-root:hover": {
-    backgroundColor: "rgb(245, 246, 253)",
+    backgroundColor: theme.palette.primary[100],
```
Comment (lucechal14) [inline frontend/workflows/projectSelector/src/project-group.tsx:97]: this is a perfect match, rgb(245,246,253) === #f5f6fd === primary[100]

File: frontend/workflows/projectSelector/src/project-links.tsx [hunk 13]
```diff
@@ -31,18 +32,18 @@ const StyledSubLink = styled.div({
...
-const StyledMoreVertIcon = styled.span({
+const StyledMoreVertIcon = styled.span(({ theme }) => ({
   ".MuiIconButton-root": {
     padding: "6px",
-    color: "rgba(theme.palette.secondary[900], 0.38)",
+    color: alpha(theme.palette.secondary[900], 0.38),
   },
   ".MuiIconButton-root:hover": {
-    backgroundColor: "rgb(245, 246, 253)",
+    backgroundColor: theme.palette.primary[100],
```
Comment (lucechal14) [inline frontend/workflows/projectSelector/src/project-links.tsx:41]: this is a perfect match, rgb(245,246,253) === #f5f6fd === primary[100]

File: frontend/workflows/projectSelector/src/project-selector.tsx [hunk 14]
```diff
@@ -65,7 +65,7 @@ const initialState: State = {
 };
 
 const StyledSelectorContainer = styled("div")(({ theme }) => ({
-  backgroundColor: "#F9FAFE",
+  backgroundColor: theme.palette.primary[50],
```
Comment (lucechal14) [inline frontend/workflows/projectSelector/src/project-selector.tsx:68]: Screenshot 2023-12-19 at 10 55 13 a m

File: frontend/workflows/projectSelector/src/project-selector.tsx [hunk 15]
```diff
@@ -96,7 +96,7 @@ const StyledProjectTextField = styled(TextField)({
 const StyledProgressContainer = styled("div")(({ theme }) => ({
   height: "4px",
   ".MuiLinearProgress-root": {
-    backgroundColor: "rgb(194, 200, 242)",
+    backgroundColor: theme.palette.primary[400],
```
Comment (lucechal14) [inline frontend/workflows/projectSelector/src/project-selector.tsx:99]: this is a perfect match, rgb(194,200,242) === #c2c8f2 === primary[400]

File: frontend/packages/core/src/icon.tsx [hunk 16]
```diff
@@ -43,12 +44,12 @@ export const StatusIcon: React.FC<StatusProps> = ({
...
         </>
       )}
       {variant === "failure" && (
         <>
-          <StyledStatusIcon data-color="#FF8A80" /> {children}
+          <StyledStatusIcon data-color={theme.palette.error[300]} /> {children}
```
Comment (lucechal14) [inline frontend/packages/core/src/icon.tsx:52]: Screenshot 2023-12-19 at 10 57 42 a m

File: frontend/workflows/envoy/src/remote-triage/clusters.tsx [hunk 17]
```diff
@@ -27,11 +28,16 @@ interface RatioStatusProps {
...
   const total = succeeded + failed;
   return (
     <>
-      {succeeded !== 0 && <Bar fill="#69F0AE" width={`${(succeeded / total) * 100}px`} />}
-      {failed !== 0 && <Bar fill="#FF8A80" width={`${(failed / total) * 100}px`} />}
+      {succeeded !== 0 && (
+        <Bar fill={theme.palette.success[200]} width={`${(succeeded / total) * 100}px`} />
+      )}
+      {failed !== 0 && (
+        <Bar fill={theme.palette.error[300]} width={`${(failed / total) * 100}px`} />
```
Comment (lucechal14) [inline frontend/workflows/envoy/src/remote-triage/clusters.tsx:39]: Screenshot 2023-12-19 at 10 58 01 a m

File: frontend/workflows/envoy/src/remote-triage/index.tsx [hunk 18]
```diff
@@ -82,12 +84,12 @@ const TriageDetails: React.FC<WizardChild> = () => {
...
       id: "Running",
       value: healthyClusterCount,
-      color: "#69F0AE",
+      color: theme.palette.success[200],
     },
     {
       id: "Failing",
       value: failingClusterCount,
-      color: "#FF8A80",
+      color: theme.palette.error[300],
```
Comment (lucechal14) [inline frontend/workflows/envoy/src/remote-triage/index.tsx:92]: Screenshot 2023-12-19 at 10 58 15 a m

**Review by jecr — APPROVED** [review]

**Pull request merged**